python/GafferUI/VectorDataWidget.py

```diff
@@ -1043,6 +1037,57 @@ def getElement( self, rowIndex, columnIndex ) :
 _DataAccessor.registerType( IECore.V3fVectorData.staticTypeId(), _CompoundDataAccessor )
 _DataAccessor.registerType( IECore.V3dVectorData.staticTypeId(), _CompoundDataAccessor )
 
+class _ColorDataAccessor( _CompoundDataAccessor ) :
+
+	def __init__( self, data ) :
+
+		_CompoundDataAccessor.__init__( self, data )
+
+	def setElement( self, rowIndex, columnIndex, value ) :
+
+		if columnIndex < self.numColumns() - 1 :
+			_CompoundDataAccessor.setElement( self, rowIndex, columnIndex, value )
+		else :
+			self.data()[rowIndex] = GafferUI._Variant.fromVariant( value )
+
+	def getElement( self, rowIndex, columnIndex ) :
+
+		if columnIndex < self.numColumns() - 1 :
+			return _CompoundDataAccessor.getElement( self, rowIndex, columnIndex )
+		else :
+			return GafferUI._Variant.toVariant( self.data()[rowIndex] )
+
+class _Color3fDataAccessor( _ColorDataAccessor ) :
+
+	def __init__( self, data ) :
+
+		_ColorDataAccessor.__init__( self, data )
+
+	def numColumns( self ) :
+
+		return 4
+
+	def headerLabel( self, columnIndex ) :
+
+		return [ "R", "G", "B", "Color" ][columnIndex]
```

Comment:
It might be nice to have no suffix at all for the swatch column, since it represents the whole value.